[Chore] Coordinator Pattern 적용

JUNY0110 opened the pull request:
## Motivation 🥳 (코드를 추가/변경하게 된 이유)
- ViewController에서 navigation, modal 등 화면 이동에 대한 역할을 덜어내기 위함.
- 메모리관리 및 앱 퍼포먼스 향상을 위함.
- EnterViewController를 ClientInfoViewController로 재사용하기 위함.
- PostingTextViewController를 EditingTextViewController로 재사용하기 위함.
- BackBarButton을 간편히 사용하기 위함

~- 보여지는 이미지 비율 수정하기 위함~

## Key Changes 🔥 (주요 구현/변경 사항)
- 전체 ViewController에 대한 `Coordinator 적용`
- `EnterViewController`, `ClientInfoViewController`에 대해 `isEnterView`를 이용해 분기처리
- `PostingTextViewController`, `EditingTextViewController`에 대해 ~`isEditView`~ `isPostView`를 이용해 분기처리
- BackBarButton extension 추가

~- HistoryView 이미지 비율 수정 (**기존 height:** `screenWidth / 4 * 3` -> **수정 height:** `(screenWidth-32) / 4 * 3`)~

## ToDo 📆 (남은 작업)
- [ ] HistoryViewController -> PostViewController 화면전환 및 코디네이터

## ScreenShot 📷 (참고 사진)

## To Reviewers 🙏 (리뷰어에게 전달하고 싶은 말)
- HistoryViewController의 대규모 수정으로, PostViewController로의 화면전환 코드는 미작성 상태입니다.
- AppCoordinator에 바로 실행가능한 `showSegmentViewController`, `showPostViewController` ~, `showSplashViewController`~ 를 주석처리해두었으니, 확인하고싶은 파트를 빠르게 선택해 확인가능합니다.
- `showPostViewController`는 `HistoryViewController에서 연결 시 삭제`될 코드입니다.
- 추가 및 수정사항이 많습니다. 하지만 비슷한 코드가 많기때문에 보는 것은 어렵지 않을 것 같네요. (어쩌다보니 868줄이네요 :))

## Reference 🔗
- [[iOS] Coordinator Pattern(feat. Swift)이란?](https://jayb-log.tistory.com/321)
- [Zedd -Coordinator Pattern](https://zeddios.medium.com/coordinator-pattern-bf4a1bc46930)
- [Coordinator Tutorial for iOS: Getting Started](https://www.kodeco.com/158-coordinator-tutorial-for-ios-getting-started)
- [삼삼하다 Repo](https://github.com/Samsamhada/MacC-Team-Samsamhada/discussions/203)
- [hlung/AppCoordinatorPattern](https://github.com/hlung/AppCoordinatorPattern/tree/feature/async-coordinator)
- [How to use the coordinator pattern in iOS apps](https://www.hackingwithswift.com/articles/71/how-to-use-the-coordinator-pattern-in-ios-apps)
- [iOS) Coordinator pattern 적용해보자- Advanced](https://gyuios.tistory.com/85)

## Close Issues 🔒 (닫을 Issue)
Close #66.

JUNY0110 committed:
Style:art:: #66 Modify variable name

- postingImage -> postImage
- 변수명 통일을 위함.

JUNY0110 committed:
Remove:fire:: #66 Delete unnecessary Coordinators

JUNY0110 committed:
Chore:wrench:: #66 Add DeveloperCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add EnterCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add LicenseCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add NoticeCoordinator file

- 공용 Delegate파일이 있음.
- popToSettingView()

JUNY0110 committed:
Chore:wrench:: #66 Add PostCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add PostImageCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add PostingCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add SettingCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add SplashCoordinator file

JUNY0110 committed:
Chore:wrench:: #66 Add presentView func

JUNY0110 committed:
Chore:wrench:: #66 Add UIBackBarButton extension

JUNY0110 committed:
Chore:wrench:: #66 Add rootViewController variable

JUNY0110 committed:
Chore:wrench:: #66 Apply EnterView Coordinator

JUNY0110 committed:
Design:iphone:: #66 Add DummyLicenseViewController

- 기본 더미뷰 생성

JUNY0110 committed:
Chore:wrench:: #66 Apply NoticeView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply DeveloperView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply PostImageView Coordinator

- 일부 attribute수정
- postImage lazy변수 제거
- postImage 기본 이미지 수정
- scrollView Background 추가

JUNY0110 committed:
Chore:wrench:: #66 Apply PostingPhoto Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply PostingText Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply SettingView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply SplashView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply SegmentView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply PostView Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply Coordinator

JUNY0110 committed:
Chore:wrench:: #66 Modify popToPostingPhotoView

- coordinator remove삭제
- protocol 수정

JUNY0110 committed:
Chore:wrench:: #66 Add presentEditingView

JUNY0110 committed:
Chore:wrench:: #66 Add PostingTextCoordinator

JUNY0110 committed:
Chore:wrench:: #66 Apply PostViewCoordinator

JUNY0110 committed:
Design:iphone:: #66 Modify SystemImage

- chevron.left -> chevron.backward

JUNY0110 committed:
Remove:fire:: #66 Delete PostingTextCoordinator

JUNY0110 committed:
Chore:wrench:: #66 Delete Access Identifier

JUNY0110 committed:
Chore:wrench:: #66 Add backBarButton

JUNY0110 committed:
Chore:wrench:: #66 Add EditingTextCoordinator

JUNY0110 committed:
Chore:wrench:: #66 Modify code

- 자잘한 수정

JUNY0110 committed:
Design:iphone:: #66 Modify View Code

- 입력뷰, 고객정보수정뷰 분기 조건 추가
- layout코드 일부 수정
- popToSettingView 프로토콜 추가

JUNY0110 committed:
Chore:wrench:: #66 Apply EditingTextCoordinator

- 코디네이터 적용
- 게시물수정 뷰, 게시물텍스트뷰 분기조건 추가

JUNY0110 committed:
Chore:wrench:: #66 Apply Coordinator

- viewController 선언부 함수 밖으로 이동
- 접근지정자 추가
- 코디네이터 적용

JUNY0110 committed:
Chore:wrench:: #66 Delete Access Identifier

JUNY0110 committed:
Style:art:: #66 Add blank

JUNY0110 committed:
Style:art:: #66 Modify variable name

- 변수명 수정(View -> ViewController)
- 네비게이션 화면전환 시 PushTo~~, 모달 화면전환은 present
- 일부 Coordinator 통합

ikanic commented on GiwazipClient/Views/HistoryViewController.swift:
```diff
@@ -131,7 +131,7 @@ extension HistoryViewController: UICollectionViewDelegate, UICollectionViewDataS
         }
 
-        return CGSize(width: screenWidth - 32, height: screenWidth / 4 * 3)
+        return CGSize(width: screenWidth - 32, height: (screenWidth - 32) / 4 * 3)
```
p3;
해당 부분은 컬렉션 뷰 전면 수정에 따라 아예 사라질 부분인데, 컨플릭트가 생길 가능성이 있을 것 같네요.

JUNY0110 replied:
> p3;
> 해당 부분은 컬렉션 뷰 전면 수정에 따라 아예 사라질 부분인데, 컨플릭트가 생길 가능성이 있을 것 같네요.

해당부분 컨플릭트 해결하면서 삭제되었습니다.

ikanic replied:
> > p3;
> > 해당 부분은 컬렉션 뷰 전면 수정에 따라 아예 사라질 부분인데, 컨플릭트가 생길 가능성이 있을 것 같네요.
> 
> 해당부분 컨플릭트 해결하면서 삭제되었습니다.

넵 확인했습니다

ikanic commented on GiwazipClient/Cells/HistoryCell.swift:
```diff
@@ -33,7 +33,7 @@ class HistoryCell: UICollectionViewCell {
     }(UILabel())
     
-    private let postImage: UIImageView = {
+    let postImage: UIImageView = {
```
이 부분은 이미 제가 필요해서 수정되었는데, 충돌은 피할 수 없겠군요.. 후 새드...

JUNY0110 replied:
> 이 부분은 이미 제가 필요해서 수정되었는데, 충돌은 피할 수 없겠군요.. 후 새드...

이 부분은 괜찮습니다 :)

ikanic replied:
> 

이미 수정되었네요.

ikanic commented on GiwazipClient/Supports/BaseCoordinator.swift:
```diff
@@ -18,6 +18,7 @@ class BaseCoordinator: Coordinator {
 
     var childCoordinators: [Coordinator] = []
 
+    var rootViewController = UINavigationController()
```
p1;
해당 부분은 바로 객체를 생성하지 않고, `var rootViewController: UINavigationController`처럼 선언하셔야 할 것 같습니다.
현재로서는 rootViewController가 사용되지 않을 때에도 무조건 객체를 생성하게 되어 있어서 불필요한 Navigation이 생기게 됩니다.

JUNY0110 replied:
> p1;
> 해당 부분은 바로 객체를 생성하지 않고, `var rootViewController: UINavigationController`처럼 선언하셔야 할 것 같습니다.
> 현재로서는 rootViewController가 사용되지 않을 때에도 무조건 객체를 생성하게 되어 있어서 불필요한 Navigation이 생기게 됩니다.

dfcd485 수정했습니다

ikanic replied:
> > p1;
> > 해당 부분은 바로 객체를 생성하지 않고, `var rootViewController: UINavigationController`처럼 선언하셔야 할 것 같습니다.
> > 현재로서는 rootViewController가 사용되지 않을 때에도 무조건 객체를 생성하게 되어 있어서 불필요한 Navigation이 생기게 됩니다.
> 
> [dfcd485](https://github.com/Samsamhada/Giwazip-Client-App/commit/dfcd485e7d94fa5334a69960c459f1e9cfab18c1) 수정했습니다

수정 확인했습니다. 다만, rootViewController가 사용되지 않을때는 충돌 위험이 있어서 !가 ?로 변경되어야할 것 같습니다.

JUNY0110 replied:
> 수정 확인했습니다. 다만, rootViewController가 사용되지 않을때는 충돌 위험이 있어서 !가 ?로 변경되어야할 것 같습니다.

rootViewController를 PostingCoordinator로 이동했습니다.
PostingCoordinator 관련 코드가 다소 수정이 되었으니 확인 바랍니다.
d8d303e

ikanic replied:
> > 수정 확인했습니다. 다만, rootViewController가 사용되지 않을때는 충돌 위험이 있어서 !가 ?로 변경되어야할 것 같습니다.
> 
> rootViewController를 PostingCoordinator로 이동했습니다. PostingCoordinator 관련 코드가 다소 수정이 되었으니 확인 바랍니다. [d8d303e](https://github.com/Samsamhada/Giwazip-Client-App/commit/d8d303e2ab3f29e1ae0d0b99f6cbde04637b0d1c)

확인했습니다. 그런데 이 부분과는 약간 별개일 수 있지만 현재 작성 마지막까지 가서 문의하기를 누르면 디코딩 에러가 발생한다는 메시지가 나오던데 괜찮은 것 일까요?

JUNY0110 replied:
> 확인했습니다. 그런데 이 부분과는 약간 별개일 수 있지만 현재 작성 마지막까지 가서 문의하기를 누르면 디코딩 에러가 발생한다는 메시지가 나오던데 괜찮은 것 일까요?

코디네이터 적용으로 ViewController간의 연결이 단절되어 생긴 문제입니다.
ViewModel 적용하게 된다면 정상 작동될 것이므로 문제없는 부분이 되겠네요.

테스트를 원하신다면, PostingTextViewController에서 `var imageDatas: [Data] = [UIImage(named: "cat")]`을 해보시면 정삭작동이 되는 것을 확인하실 수 있습니다.

1758614
추가로, 임시 viewModel변수로 사용되던 networkManager를 postingTextViewController에서 networkManager 사용으로 수정하였습니다.

ikanic replied:
> > 확인했습니다. 그런데 이 부분과는 약간 별개일 수 있지만 현재 작성 마지막까지 가서 문의하기를 누르면 디코딩 에러가 발생한다는 메시지가 나오던데 괜찮은 것 일까요?
> 
> 코디네이터 적용으로 ViewController간의 연결이 단절되어 생긴 문제입니다. ViewModel 적용하게 된다면 정상 작동될 것이므로 문제없는 부분이 되겠네요.
> 
> 테스트를 원하신다면, PostingTextViewController에서 `var imageDatas: [Data] = [UIImage(named: "cat")]`을 해보시면 정삭작동이 되는 것을 확인하실 수 있습니다.
> 
> [1758614](https://github.com/Samsamhada/Giwazip-Client-App/commit/1758614a06e0052d05c7d706c16e5a4b9e55081f) 추가로, 임시 viewModel변수로 사용되던 networkManager를 postingTextViewController에서 networkManager 사용으로 수정하였습니다.

확인했습니다. 리뷰 리졸브합니다~

ikanic commented on GiwazipClient/Coordinators/AppCoordinator.swift:
```diff
@@ -5,66 +5,145 @@
+//            showPostViewController()
         } else {
-            showMainViewController()
+            showSegmentViewController()
```
p1;
해당 부분 조건이 반대로 된 것 같습니다.
isLoggenIn은 로그인 된 상태이므로 해당 부분에 segment가 와야할 것 같네요.
그리고 주석은 후에 사용하려고 삭제하지 않으신 걸까요?

JUNY0110 replied:
> p1;
> 해당 부분 조건이 반대로 된 것 같습니다.
> isLoggenIn은 로그인 된 상태이므로 해당 부분에 segment가 와야할 것 같네요.
> 그리고 주석은 후에 사용하려고 삭제하지 않으신 걸까요?

음, 해당부분은 PR에도 작성되어있다시피 모두 편하게 테스트하도록 일부러 isLoggedIn에 코드를 남겨둔 부분입니다.
일단 HistoryVC가 전반적으로 수정된지 얼마되지 않아 
HistoryViewController와 PostViewController가 연결하지 않았기때문에 더더욱 그렇게 뒀습니다.
다만, else 밑에 있는 showSegment는 착각한 부분이 맞네요 수정했습니다. a246a88

ikanic replied:
> > p1;
> > 해당 부분 조건이 반대로 된 것 같습니다.
> > isLoggenIn은 로그인 된 상태이므로 해당 부분에 segment가 와야할 것 같네요.
> > 그리고 주석은 후에 사용하려고 삭제하지 않으신 걸까요?
> 
> 음, 해당부분은 PR에도 작성되어있다시피 모두 편하게 테스트하도록 일부러 isLoggedIn에 코드를 남겨둔 부분입니다. 일단 HistoryVC가 전반적으로 수정된지 얼마되지 않아 HistoryViewController와 PostViewController가 연결하지 않았기때문에 더더욱 그렇게 뒀습니다. 다만, else 밑에 있는 showSegment는 착각한 부분이 맞네요 수정했습니다. [a246a88](https://github.com/Samsamhada/Giwazip-Client-App/commit/a246a881fe9af9ea02db4e2af2006a4648152b54)

그렇다면 isLoggedIn 분기를 이렇게 하는 거보다는 isLoggedIn 자체의 값을 false로 변경해주는게 더 좋을 것 같네요.

JUNY0110 replied:
> > > p1;
> > > 해당 부분 조건이 반대로 된 것 같습니다.
> > > isLoggenIn은 로그인 된 상태이므로 해당 부분에 segment가 와야할 것 같네요.
> > > 그리고 주석은 후에 사용하려고 삭제하지 않으신 걸까요?
> > 
> > 
> > 음, 해당부분은 PR에도 작성되어있다시피 모두 편하게 테스트하도록 일부러 isLoggedIn에 코드를 남겨둔 부분입니다. 일단 HistoryVC가 전반적으로 수정된지 얼마되지 않아 HistoryViewController와 PostViewController가 연결하지 않았기때문에 더더욱 그렇게 뒀습니다. 다만, else 밑에 있는 showSegment는 착각한 부분이 맞네요 수정했습니다. [a246a88](https://github.com/Samsamhada/Giwazip-Client-App/commit/a246a881fe9af9ea02db4e2af2006a4648152b54)
> 
> 그렇다면 isLoggedIn 분기를 이렇게 하는 거보다는 isLoggedIn 자체의 값을 false로 변경해주는게 더 좋을 것 같네요.

1af26db
isLoggedIn만 사용해서 테스트작업하도록 하는게 편할거라 생각했는데, Merge되는 내용이니 조금 귀찮아지더라도 정상적으로 보내는게 낫겠네요.

ikanic replied:
> > > > p1;
> > > > 해당 부분 조건이 반대로 된 것 같습니다.
> > > > isLoggenIn은 로그인 된 상태이므로 해당 부분에 segment가 와야할 것 같네요.
> > > > 그리고 주석은 후에 사용하려고 삭제하지 않으신 걸까요?
> > > 
> > > 
> > > 음, 해당부분은 PR에도 작성되어있다시피 모두 편하게 테스트하도록 일부러 isLoggedIn에 코드를 남겨둔 부분입니다. 일단 HistoryVC가 전반적으로 수정된지 얼마되지 않아 HistoryViewController와 PostViewController가 연결하지 않았기때문에 더더욱 그렇게 뒀습니다. 다만, else 밑에 있는 showSegment는 착각한 부분이 맞네요 수정했습니다. [a246a88](https://github.com/Samsamhada/Giwazip-Client-App/commit/a246a881fe9af9ea02db4e2af2006a4648152b54)
> > 
> > 
> > 그렇다면 isLoggedIn 분기를 이렇게 하는 거보다는 isLoggedIn 자체의 값을 false로 변경해주는게 더 좋을 것 같네요.
> 
> [1af26db](https://github.com/Samsamhada/Giwazip-Client-App/commit/1af26db23e7656c692fc84f0d38fbddac1753394) isLoggedIn만 사용해서 테스트작업하도록 하는게 편할거라 생각했는데, Merge되는 내용이니 조금 귀찮아지더라도 정상적으로 보내는게 낫겠네요.

확인했습니다. 리뷰 리졸브합니다.

JUNY0110 committed:
Chore:wrench:: #66 Add workEnding

- 시공마감버튼 클릭 시 확인메시지 호출
- 확인 버튼 클릭 시 SplashViewController로 이동
- SplashViewController 호출방식 수정

JUNY0110 committed:
Style:art:: #66 Modify variable name

- SegmentViewController 버튼 클릭에 대한 함수명 수정
- showPostingPhotoViewController -> didTapInquiryButton
- showSettingViewController -> didTapSettingButton

JUNY0110 committed:
Fix:beetle:: Clear pbx Error

- pbx 에러 수정

JUNY0110 committed:
Fix:beetle:: #66 Add LineDashPattern

- LineDashPattern Extension 복구

JUNY0110 committed:
Rename:pencil:: #66 Modify file name

JUNY0110 committed:
Chore:wrench:: #66 Delete viewControllers

- navigationController 뷰컨트롤러가 쌓이지 않도록 방지하기 위함.
- Splash -> Enter -> Segment 화면전환 시 Splash, Enter 쌓여있는 ViewControllers 삭제

JUNY0110 committed:
Design:iphone:: #66 Add viewDidLoad Inheritance

JUNY0110 committed:
Style:art:: #66 Add Remarks

JUNY0110 committed:
Style:art:: #66 Add Remarks

ikanic commented on GiwazipClient/Coordinators/AppCoordinator.swift:
```diff
@@ -5,66 +5,145 @@
 
-    private func showSubViewController() {
-        let coordinator = SubCoordinator(navigationController: navigationController)
+    func showSplashViewController() {
+        let coordinator = SplashCoordinator(navigationController: navigationController)
+        coordinator.delegate = self
+        coordinator.start()
+        self.childCoordinators.append(coordinator)
+    }
+
+    private func showSegmentViewController() {
+        let coordinator = SegmentCoordinator(navigationController: navigationController)
+        coordinator.delegate = self
+        coordinator.start()
+        self.childCoordinators.append(coordinator)
+    }
```
이 친구들의 반복되는 부분을 합칠 아이디어가 하나 있긴 합니다만, 아직 코디네이터에 대한 이해가 부족해서 완전한 동작인지 불확실하네요.
후에 완전하게 동작 가능하다면 공유하겠습니다.

ikanic commented on GiwazipClient/Coordinators/EditingTextCoordinator.swift:
```diff
@@ -0,0 +1,44 @@
+        
+        postingTextViewController.textView.text = postViewController.postingDescription.text
+        postingTextViewController.isEditView = true
+//
```
p1;
요 주석은 뭘까요

JUNY0110 replied:
> p1;
> 요 주석은 뭘까요

08ba416 시행착오의 산물이네요. 확인감사합니다.

ikanic replied:
> > p1;
> > 요 주석은 뭘까요
> 
> [08ba416](https://github.com/Samsamhada/Giwazip-Client-App/commit/08ba4161e8f96c7e66129812f5dbb7f0196005f7) 시행착오의 산물이네요. 확인감사합니다.

확인했습니다. 리뷰 리졸브합니다.

ikanic commented on GiwazipClient/Coordinators/EnterCoordinator.swift:
```diff
@@ -0,0 +1,36 @@
+    // MARK: - Property
+    
+    var delegate: EnterCoordinatorDelegate?
+    let enterViewController = EnterViewController()
```
Q.
얜 왜 여기에서 초기화되고 있는 걸까요?

JUNY0110 replied:
> Q.
> 얜 왜 여기에서 초기화되고 있는 걸까요?

AppCoordinator에서 조건분기를 위함입니다.
```swift
    func pushToEnterViewController() {
        let coordinator = EnterCoordinator(navigationController: navigationController)
        coordinator.enterViewController.isEnterView = true    // <--
        coordinator.delegate = self
        coordinator.start()
        self.childCoordinators.append(coordinator)
    }

    func pushToClientInfoViewController() {
        let coordinator = EnterCoordinator(navigationController: navigationController)
        coordinator.enterViewController.isEnterView = false    // <--
        coordinator.delegate = self
        coordinator.start()
        self.childCoordinators.append(coordinator)
    }
```

ikanic replied:
> > Q.
> > 얜 왜 여기에서 초기화되고 있는 걸까요?
> 
> AppCoordinator에서 조건분기를 위함입니다.
> 
> ```swift
>     func pushToEnterViewController() {
>         let coordinator = EnterCoordinator(navigationController: navigationController)
>         coordinator.enterViewController.isEnterView = true    // <--
>         coordinator.delegate = self
>         coordinator.start()
>         self.childCoordinators.append(coordinator)
>     }
> 
>     func pushToClientInfoViewController() {
>         let coordinator = EnterCoordinator(navigationController: navigationController)
>         coordinator.enterViewController.isEnterView = false    // <--
>         coordinator.delegate = self
>         coordinator.start()
>         self.childCoordinators.append(coordinator)
>     }
> ```

조건 분기를 위함이라면 위에서 EnterViewController를 초기화하는 것보다는 func의 파라미터로 bool 값을 받는 것이 더 옳을 것 같습니다. 우선 저기서 생성한 EnterViewController가 기존의 EnterViewController와 같은 뷰가 아닐 것이기 때문에 더더욱 이 부분은 변경되어야 할 것 같네요.

ikanic replied:
> > Q.
> > 얜 왜 여기에서 초기화되고 있는 걸까요?
> 
> AppCoordinator에서 조건분기를 위함입니다.
> 
> ```swift
>     func pushToEnterViewController() {
>         let coordinator = EnterCoordinator(navigationController: navigationController)
>         coordinator.enterViewController.isEnterView = true    // <--
>         coordinator.delegate = self
>         coordinator.start()
>         self.childCoordinators.append(coordinator)
>     }
> 
>     func pushToClientInfoViewController() {
>         let coordinator = EnterCoordinator(navigationController: navigationController)
>         coordinator.enterViewController.isEnterView = false    // <--
>         coordinator.delegate = self
>         coordinator.start()
>         self.childCoordinators.append(coordinator)
>     }
> ```

아 제가 착각했네요. 이 부분은 이게 맞는 것 같네요. 혼란 드려서 죄송합니다~ 리뷰 리졸브합니다~

ikanic commented on GiwazipClient/Views/EnterViewController.swift:
```diff
@@ -66,79 +75,99 @@ class EnterViewController: BaseViewController {
-        $0.backgroundColor = .gray
-        $0.layer.cornerRadius = 16
+        $0.configuration?.title = TextLiteral.enterButtonText
+        $0.configuration?.attributedTitle?.font = UIFont.systemFont(ofSize: 18, weight: .bold)
```
p3;
버튼이 변경됨에 따라 세그먼트 뷰와 같은 버튼인 것 같은데, 그렇다면 weight가 bold가 아니라 semibold가 되어야 할 것 같네요.
피그마 상에 수정되지 않은 16 bold와 수정 후인 18 semibold가 섞여있어서 헷갈리신 것 같습니다.

JUNY0110 replied:
> p3;
> 버튼이 변경됨에 따라 세그먼트 뷰와 같은 버튼인 것 같은데, 그렇다면 weight가 bold가 아니라 semibold가 되어야 할 것 같네요.
> 피그마 상에 수정되지 않은 16 bold와 수정 후인 18 semibold가 섞여있어서 헷갈리신 것 같습니다.

PostingTextVC의 InquiryButton이 18 bold를 사용하고 있어 착각했네요.
EnterVC와 PostingTextVC 둘 다 18 semibold로 수정했습니다.
204e0b1

ikanic replied:
> 

확인했습니다. 그런데 알러트 조건 분기 부분이 주석처리된 이유가 있을까요?

JUNY0110 replied:
> 확인했습니다. 그런데 알러트 조건 분기 부분이 주석처리된 이유가 있을까요?

이 또한 작업 편하게 하도록 주석한 것이었지만 되살렸습니다.
7223b59

ikanic replied:
> > 확인했습니다. 그런데 알러트 조건 분기 부분이 주석처리된 이유가 있을까요?
> 
> 이 또한 작업 편하게 하도록 주석한 것이었지만 되살렸습니다. [7223b59](https://github.com/Samsamhada/Giwazip-Client-App/commit/7223b59ec667a7893a18b08660fca722cd623d92)

확인했습니다. 리뷰 리졸브합니다.

ikanic commented on GiwazipClient/Views/EnterViewController.swift:
```diff
@@ -66,79 +75,99 @@ class EnterViewController: BaseViewController {
+
+        if isEnterView {
+            enterButton.configuration?.title = TextLiteral.enterButtonText
+            inviteCodeInput.isHidden = false
+            inviteCodeUnderLine.isHidden = false
+            inviteCodeGuideLabel.isHidden = false
```
p5;
이렇게 많은 부분이 hidden되어야 한다면, 하나의 컨테이너 뷰로 묶어서 컨테이너 뷰만 hidden 시키는 것이 코드 가독성 면에서는 더 좋을 것 같네요.

JUNY0110 replied:
> p5;
> 이렇게 많은 부분이 hidden되어야 한다면, 하나의 컨테이너 뷰로 묶어서 컨테이너 뷰만 hidden 시키는 것이 코드 가독성 면에서는 더 좋을 것 같네요.

f7da0f9
view의 constraint를 모두 잡지 않으면 내부 view에 대한 동작이 안 되는 거였네요.

ikanic replied:
> f7da0f9

넵넵 맞습니다. 수정 확인했습니다. 리뷰 리졸브합니다~

ikanic commented on GiwazipClient/Views/EnterViewController.swift:
```diff
@@ -152,14 +181,14 @@ class EnterViewController: BaseViewController {
     @objc func checkPhoneNumberLength() {
-        guard let phoneNumber = phoneNumberInput.text?.replacingOccurrences(of: " - ", with: "")
-        else { return }
+        guard let phoneNumber = phoneNumberInput.text?.replacingOccurrences(of: " - ", with: "") else { return }
```
p3;
한 라인이 100자를 넘어가네요. 이 부분은 라인을 나누기 어려운 부분도 아니므로 else 앞에서 한줄 띄어주면 좋을 것 같습니다.

JUNY0110 replied:
> p3;
> 한 라인이 100자를 넘어가네요. 이 부분은 라인을 나누기 어려운 부분도 아니므로 else 앞에서 한줄 띄어주면 좋을 것 같습니다.

88cf9a7 줄바꿈 했습니다.

ikanic replied:
> > p3;
> > 한 라인이 100자를 넘어가네요. 이 부분은 라인을 나누기 어려운 부분도 아니므로 else 앞에서 한줄 띄어주면 좋을 것 같습니다.
> 
> [88cf9a7](https://github.com/Samsamhada/Giwazip-Client-App/commit/88cf9a7ba0602460803359a7a4299b3ea5b50e2e) 줄바꿈 했습니다.

확인했습니다. 리뷰 리졸브합니다.

ikanic commented on GiwazipClient/Coordinators/EditingTextCoordinator.swift:
```diff
@@ -0,0 +1,44 @@
+    // MARK: - Method
+    
+    override func start() {
+        let postingTextViewController = PostingTextViewController()
+        let postViewController = PostViewController()
+        
+        postingTextViewController.delegate = self
+        
+        postingTextViewController.textView.text = postViewController.postingDescription.text
+        postingTextViewController.isEditView = true
+//
+        rootViewController = UINavigationController(rootViewController: postingTextViewController)
+        rootViewController.modalPresentationStyle = .fullScreen
+
+        navigationController.present(rootViewController, animated: true)
```
이 부분은 뭔가 이상하다는 생각이 듭니다만, 현재 확인이 불가능해서 ㅠㅠ

taek0622 reviewed (changes requested):
고생하셨습니다.
생각보다 많이 복잡하네요.
일단 보이는 부분에 리뷰 남겼고, 가장 큰 문제는 현재 HistoryCoordinator가 넘어오지 않아서 아예 빌드가 되지 않습니다. 확인 하시고 HistoryCoordinator 파일도 넣어주세요.

JUNY0110 committed:
Chore:wrench:: #66 Modify loggedIn showVC

JUNY0110 committed:
Style:art:: #66 Uneccessary remarks

JUNY0110 committed:
Chore:wrench:: #66 Modify font weight

JUNY0110 committed:
Style:art:: #66 Add inviteContainer

JUNY0110 committed:
Style:art:: #66 Modify spacing

JUNY0110 committed:
Chore:wrench:: #66 Modify rootViewController

- 선언이 아닌 타입으로 변경
- 미사용 객체 상주 방지

JUNY0110 committed:
Chore:wrench:: #66 Delete unused code

JUNY0110 committed:
Chore:wrench:: #66 Delete unused code

JUNY0110 committed:
Design:iphone:: #66 Modify button color

JUNY0110 committed:
Style:art:: #66 Modify protocol func name

JUNY0110 committed:
Chore:wrench:: #66 Add backbutton argument

JUNY0110 committed:
Chore:wrench:: #66 Modify PostingTextCoordinator

- EditingTextCoordinator -> PostingTextCoordinator로 수정
- isPostTextView로 조건 분기

JUNY0110 committed:
Chore:wrench:: #66 Modify PostingCoordinator

- postingTextCoordinator를 coordinator로 호출하여, PostingCoordinator를 SubAppCoordinator로 활용
- rootViewController를 PostingCoordinator로 이동

JUNY0110 committed:
Designe:iphone:: #66 Add isPostTextView condition

- isPostTextView 조건 추가에 따른 navigation 및 버튼 수정

JUNY0110 committed:
Chore:wrench:: #66 Add HistoryCoordinator

JUNY0110 committed:
Chore:wrench:: #66 Modify AppCoordinator condition

JUNY0110 committed:
Chore:wrench:: #66 Modify code

- EnterViewController 알림 코드 되살림

taek0622 reviewed (approved):
고생하셨습니다.
원래도 큰 PR이었지만, 리베이스에 수정에 하다보니 868줄이나 되어버렸네요.
현재 post를 클릭했을 때 동작이 되지 않지만, 해당 부분은 후에 제가 다시 한 번 살펴보도록 하겠습니다.
추가로 디코딩 에러 발생에 대해서 질문 하나 남겼는데 확인하시고 머지해주시면 감사하겠습니다.

JUNY0110 committed:
Chore:wrench:: #66 Modify textView text

- 기본값 추가
- networkmanager 사용
